abstractions/php/src/Serialization/SerializationWriter.php

```diff
@@ -0,0 +1,142 @@
+<?php
+namespace Microsoft\Kiota\Abstractions\Serialization;
+
+use DateInterval;
+use DateTime;
+use Microsoft\Kiota\Abstractions\Enum;
+use Psr\Http\Message\StreamInterface;
+
+/** Defines an interface for serialization of objects to a stream. */
+interface SerializationWriter {
+    /**
+     * Writes the specified string value to the stream with an optional given key.
+     * @param string $key the key to write the value with.
+     * @param string $value the value to write to the stream.
+     */
+    public function writeStringValue(string $key, string $value): void;
+
+    /**
+     * Writes the specified Boolean value to the stream with an optional given key.
+     * @param string $key the key to write the value with.
+     * @param bool $value the value to write to the stream.
+     */
+    public function writeBooleanValue(string $key, bool $value): void;
```

Comment:
are primitive values nullable in PHP? If yes we're ok, otherwise we need to use the nullable type equivalent. There's a semantic difference between false and null (not set)

Reply:
Hi @baywet, yes primitive types are nullable but you have to explicitly state it by writing `?type $name` instead of `type $name`.